### PR TITLE
gh-108113: Make it possible to create an optimized AST

### LINKED ISSUE #108113: compile() with PyCF_ONLY_AST flag ignores the optimize arg
``compile()`` (or ``ast.parse()``, which calls it) can return the ast for some source code, converted to a Python object version of the AST. But this AST is not optimized (with ``_PyAST_Optimize``). 

Static Python re-implemented ``_PyAST_Optimize`` for this reason. If we expose a way for them to get an optimized AST, they won't have to.

For performance, it would be better not to call another API function on a Python AST, but to add an API option that runs ``_PyAST_Optimize`` before converting the AST to Python.

### Linked PRs
* gh-108154
* gh-108282
* gh-113241

### PR DESCRIPTION
**Pull request by iritkatriel**
Fixes #108113.

* Issue: gh-108113

----
:books: Documentation preview :books:: https://cpython-previews--108154.org.readthedocs.build/

### CODE REVIEW

File: Doc/whatsnew/3.13.rst [hunk 1]
```diff
@@ -85,6 +85,12 @@ Other Language Changes
   This change will affect tools using docstrings, like :mod:`doctest`.
   (Contributed by Inada Naoki in :gh:`81283`.)
 
+* The :func:`compile` built-in can now accept a new flag,
+  ``ast.PyCF_OPTIMIZED_AST``, which is similar to ``ast.PyCF_ONLY_AST``
```
Comment (corona10) [inline Doc/whatsnew/3.13.rst:89]:
```suggestion
  :data:`ast.PyCF_OPTIMIZED_AST`, which is similar to :data:`ast.PyCF_ONLY_AST`
```

File: Lib/test/test_builtin.py [hunk 2]
```diff
@@ -369,6 +369,7 @@ def f(): """doc"""
                   (1, False, 'doc', False, False),
                   (2, False, None, False, False)]
         for optval, *expected in values:
+          with self.subTest(opt = optval):
```
Comment (corona10) [inline Lib/test/test_builtin.py]: Wait 2 spaces?

**Review by carljm — COMMENTED** [review]
Looks like there's a docs build issue, but otherwise this LGTM.

**Review by corona10 — APPROVED** [review]

**Comment by dreamflow** [comment]
should the new `ast.PyCF_OPTIMIZED_AST`
be mentioned in the [compiler-flags](https://docs.python.org/3.13/library/ast.html#compiler-flags) doc ?

the [compile()](https://docs.python.org/3.13/library/functions.html#compile) doc has a link to the [compiler-flags](https://docs.python.org/3.13/library/ast.html#compiler-flags) doc ,
so that someone reading the [compile()](https://docs.python.org/3.13/library/functions.html#compile) doc , can find out
that compile() has the `ast.PyCF_OPTIMIZED_AST` option .

**Comment by iritkatriel** [comment]
@dreamflow Would you like to create a PR for that doc change?

**Comment by dreamflow** [comment]
@iritkatriel  yes , i'm happy to 🙂

have posted the PR https://github.com/python/cpython/pull/113241

**Pull request merged**